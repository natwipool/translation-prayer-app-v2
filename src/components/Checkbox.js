import React from 'react';

const Checkbox = props => {
  const onCheckboxChange = e => {
    const list = e.target.value;

    if (e.target.checked) {
      props.onAddLists(list);
    } else if (!e.target.checked) {
      props.onRemoveLists(list);
    }
  };

  return (
    <div>
      <label htmlFor={props.precept} className="checkbox">
        <input
          type="checkbox"
          id={props.precept}
          checked={props.lists.find(list => list === props.precept)}
          value={props.precept}
          onChange={onCheckboxChange}
        />
        <div className="checkbox-content">
        <span />
          
            <p className="checkbox-content__title">{props.precept}</p>
            
            {
              props.lists.indexOf(props.precept) > -1 &&
              <p className="checkbox-content__number">{props.lists.indexOf(props.precept) + 1}</p>
            } 
          
        </div>
      </label>
    </div>
  );
};

export default Checkbox;
