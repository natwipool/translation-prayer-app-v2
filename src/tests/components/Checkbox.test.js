import React from 'react';
import { shallow } from 'enzyme';
import Checkbox from '../../components/Checkbox';
import transPrayers from '../fixtures/transPrayers';

test('should render Checkbox correctly', () => {
  const wrapper = shallow(<Checkbox {...transPrayers[0]} lists={[]} />);
  expect(wrapper).toMatchSnapshot();
});

test('should call onAddLists with value if input check changed', () => {
  const onAddListsSpy = jest.fn();
  const value = transPrayers[0].precept;
  const wrapper = shallow(<Checkbox
    {...transPrayers[0]}
    lists={[]}
    onAddLists={onAddListsSpy}
  />);
  wrapper.find('input').simulate('change', {
    target: {
      checked: true,
      value,
    }
  });
  expect(onAddListsSpy).toHaveBeenLastCalledWith(value);
});

test('should call onRemoveLists with value if input uncheck changed', () => {
  const onRemoveListsSpy = jest.fn();
  const value = transPrayers[0].precept;
  const wrapper = shallow(<Checkbox
    {...transPrayers[0]}
    lists={[]}
    onRemoveLists={onRemoveListsSpy}
  />);
  wrapper.find('input').simulate('change', {
    target: {
      checked: false,
      value,
    }
  });
  expect(onRemoveListsSpy).toHaveBeenLastCalledWith(value);
});