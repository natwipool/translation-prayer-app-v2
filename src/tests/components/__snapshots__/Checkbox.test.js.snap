// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Checkbox correctly 1`] = `
<div>
  <input
    id="เขมาเขมสรณคมนปริทีปิกาคาถา"
    onChange={[Function]}
    type="checkbox"
    value="เขมาเขมสรณคมนปริทีปิกาคาถา"
  />
  <label
    htmlFor="เขมาเขมสรณคมนปริทีปิกาคาถา"
  >
    <b>
      เขมาเขมสรณคมนปริทีปิกาคาถา
    </b>
     - 
    บทสวดมนต์ภาคเย็น
     
  </label>
</div>
`;
